File: user-service/src/main/java/org/aston/ems/user_service/service/CustomDigitComparatorReverse.java
package org.aston.ems.user_service.service;

import org.aston.ems.user_service.dto.UserDTO;
import org.springframework.stereotype.Service;

import java.util.Comparator;

@Service
public class CustomDigitComparatorReverse implements Comparator<UserDTO> {
    @Override
    public int compare(UserDTO o1, UserDTO o2) {
        return Integer.compare(0, o1.getMark() - o2.getMark());
    }
}

File: user-service/src/main/java/org/aston/ems/user_service/service/CustomDigitComparatorReverse.kt
package org.aston.ems.user_service.service

import org.aston.ems.user_service.dto.UserDTO
import org.springframework.stereotype.Service

@Service
class CustomDigitComparatorReverse : Comparator<UserDTO> {
    override fun compare(o1: UserDTO, o2: UserDTO): Int {
        return 0.compareTo(o1.mark- o2.mark)
    }
}
